### PR TITLE
Subtitles support for VideoView.

### PR DESCRIPTION
**Pull request by okrasnob**
Added the selection of the TrackGroup by specifying the groupIndex in the renderer's TrackGroupArray.

To enable subtitles, call:
videoView.setRendererTrackGroupIndex(ExoMedia.RendererType.CLOSED_CAPTION, 0);
where '0' is the groupIndex in the subtitles renderer tracks.

To disable:
videoView.setRendererTrackGroupIndex(ExoMedia.RendererType.CLOSED_CAPTION, ExoMedia.TRACK_GROUP_DISABLED);

To get the currently selected track group index (if any) in the specified renderer:
videoView.getRendererTrackGroupIndex(ExoMedia.RendererType.CLOSED_CAPTION);

###### Addresses issue #115.
- [ ] This pull request follows the coding standards

###### This PR changes:
 -

### CODE REVIEW

File: library/src/main/java/com/devbrackets/android/exomedia/core/api/VideoViewApi.java [hunk 1]
```diff
@@ -112,6 +112,10 @@
 
     void setTrack(ExoMedia.RendererType type, int trackIndex);
 
+    void setRendererTrackGroupIndex(ExoMedia.RendererType trackType, int trackGroupIndex);
```
Comment (brianwernick) [inline library/src/main/java/com/devbrackets/android/exomedia/core/api/VideoViewApi.java]: This looks like it's a duplicate of `setTrack`. Why did you need to define a separate function for this?

File: library/src/main/res/layout/exomedia_video_view_layout.xml [hunk 2]
```diff
@@ -17,4 +17,9 @@
         android:layout_height="match_parent"
         android:scaleType="centerCrop"
         tools:ignore="ContentDescription"/>
+
+    <com.google.android.exoplayer2.ui.SubtitleView
```
Comment (brianwernick) [inline library/src/main/res/layout/exomedia_video_view_layout.xml]: I've purposefully not included the UI package, we should handle this on our own instead of depending on the one they provide (that is specifically created for their simple exo player)

**Comment by brianwernick** [comment]
The main setback for me is that previously I was of the opinion that this should be compatible with the systems MediaPlayer subtitle implementation which is why it wasn't already done (see https://github.com/brianwernick/ExoMedia/commit/8f55507d686a1e0efb6439968c9c726033fd4c2c) but at this point where the fallback rate is pretty low I'm not sure that's necessary anymore.

File: library/src/main/java/com/devbrackets/android/exomedia/ExoMedia.java [hunk 3]
```diff
@@ -37,6 +37,8 @@
         METADATA
     }
 
+    public static final int TRACK_GROUP_DISABLED  = -1;
```
Comment (brianwernick) [inline library/src/main/java/com/devbrackets/android/exomedia/ExoMedia.java]: This doesn't belong here, at the very least it belongs in the ExoMediaPlayer class

File: library/src/main/java/com/devbrackets/android/exomedia/core/api/VideoViewApi.java (hunk 1 above)
Comment (okrasnob) [inline library/src/main/java/com/devbrackets/android/exomedia/core/api/VideoViewApi.java]: void setTrack(...) is causing the crash with "index out of bounds" exception when trying to select the first subtitle track (which is present in the stream):
`setTrack(ExoMedia.RendererType.CLOSED_CAPTION, 0);
When looking at the  "setTrack" implementation I have noticed that it might be used to select the different bandwidth tracks for video/audio.
Shall I change "setTrack" to make it work with the non-adaptive tracks instead?

File: library/src/main/res/layout/exomedia_video_view_layout.xml (hunk 2 above)
Comment (okrasnob) [inline library/src/main/res/layout/exomedia_video_view_layout.xml]: Shall I duplicate the SubtitleView logic?

File: library/src/main/java/com/devbrackets/android/exomedia/ExoMedia.java (hunk 3 above)
Comment (okrasnob) [inline library/src/main/java/com/devbrackets/android/exomedia/ExoMedia.java]: Ok, I will change it. Thank you.

File: library/src/main/res/layout/exomedia_video_view_layout.xml (hunk 2 above)
Comment (brianwernick) [inline library/src/main/res/layout/exomedia_video_view_layout.xml]: Yeah, lets do that. Thanks

File: library/src/main/java/com/devbrackets/android/exomedia/core/api/VideoViewApi.java (hunk 1 above)
Comment (brianwernick) [inline library/src/main/java/com/devbrackets/android/exomedia/core/api/VideoViewApi.java]: Yes. The original intent was to have it work for both the adaptive tracks as well as things like CC.
